clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkThread.java

```diff
@@ -192,10 +191,11 @@ static void runAtClose(final Collection<Optional<? extends RequestManager>> requ
 
         // Poll to ensure that request has been written to the socket. Wait until either the timer has expired or until
         // all requests have received a response.
-        while (timer.notExpired() && !requestFutures.stream().allMatch(Future::isDone)) {
-            networkClientDelegate.poll(timer.remainingMs(), timer.currentTimeMs());
+        do {
```

Comment:
We need to make sure to at least poll the network client once, so switched to do {} while